src/views/ModelPlan/Documents/AddDocument/LinkDocument.tsx

```diff
@@ -0,0 +1,416 @@
+                              label={translateDocumentType(
+                                documentType as DocumentType
+                              )}
+                              onChange={() => {
+                                setFieldValue('documentType', documentType);
+                                setFieldValue('otherTypeDescription', '');
+                              }}
+                              value={documentType}
+                            />
+                          );
+                        })}
+                      <Field
+                        as={RadioField}
+                        checked={values.documentType === 'OTHER'}
+                        id="FileUpload-OTHER"
+                        name="documentType"
+                        label={translateDocumentType(DocumentType.OTHER)}
+                        value="OTHER"
+                      />
+                      {values.documentType === 'OTHER' && (
+                        <div className="margin-left-4 margin-bottom-1">
+                          <FieldGroup
+                            scrollElement="otherTypeDescription"
+                            error={!!flatErrors.otherTypeDescription}
+                          >
+                            <Label
+                              htmlFor="FileUpload-OtherType"
+                              className="margin-bottom-1"
+                            >
+                              {t('documentKind')}
+                              <RequiredAsterisk />
+                            </Label>
+                            <FieldErrorMsg>
+                              {flatErrors.otherTypeDescription}
+                            </FieldErrorMsg>
+                            <Field
+                              as={TextField}
+                              error={!!flatErrors.otherTypeDescription}
+                              className="margin-top-0"
+                              id="FileUpload-OtherTypeDescription"
+                              name="otherTypeDescription"
+                            />
+                          </FieldGroup>
+                        </div>
+                      )}
+                    </fieldset>
+                  </FieldGroup>
+
+                  <FieldGroup
+                    scrollElement="restricted"
+                    error={!!flatErrors.restricted}
+                  >
+                    <Label
+                      htmlFor="document-upload-restricted-yes"
+                      className="maxw-none"
+                    >
+                      {t('costQuestion')}
+                      <RequiredAsterisk />
+                    </Label>
+
+                    <p className="margin-0 line-height-body-4">
+                      {t('costInfo')}
+                    </p>
+
+                    <FieldErrorMsg>{flatErrors.restricted}</FieldErrorMsg>
+                    <Fieldset>
+                      {[true, false].map(key => (
```

Comment:
We have a comonent for rendering basic boolean radio `<BooleanRadio />`.  I think in the normal document upload was created before this reusuable component was created, but just wanted to throw this out there as it may condense code a bit.